firmware/wheel.ino

```diff
@@ -0,0 +1,43 @@
+#define enA 9
+#define in1 6
+#define in2 7
+#define button 4
+
+int rotDirection = 0;
+int pressed = false;
+
+void setup() {
+    pinMode(enA, OUTPUT);
+    pinMode(in1, OUTPUT);
+    pinMode(in2, OUTPUT);
+    pinMode(button, INPUT);
+    digitalWrite(in1, LOW);
+    digitalWrite(in2, HIGH);
+}
+
+void loop() {
+    int potValue = analogRead(A0);
+    int pwmOutput = map(potValue, 0, 1023, 0, 255);
+    analogWrite(enA, pwmOutput);
+
+    if (digitalRead(button) == true) {
+        pressed = !pressed;
+    }
+
+    while (digitalRead(button) == true);
+        delay(20);
```

Comment:
how come the delays are so arbitrary between 20 100 and 1000? are they actaully becaus they're diff types of pins? does it make sense to map them to some shared constants? or is it less performant